pkg/util/enc.go

```diff
@@ -55,3 +55,22 @@ func TryToDecodeBase64Gzip(data []byte) (string, error) {
 
 	return string(plainText), nil
 }
+
+// EncodeGzipBase64 compresses the input string using gzip and then encodes it
+// using base64.
+func EncodeGzipBase64(s string) (string, error) {
+	var zbuf bytes.Buffer
+	zw := gzip.NewWriter(&zbuf)
+	if _, err := zw.Write([]byte(s)); err != nil {
+		return "", err
+	}
+	if err := zw.Flush(); err != nil {
```

Comment:
I was just moving this function from `session_utils.go` to this `util` package that's more generic. I didn't notice it had an extra flush but after checking out the source code, looks like it's redundant indeed. Gonna remove it and update the test accordingly.

Reply:
I added the `flush()` call back because some test specs failed due to hardcoded expected strings that didn't match without it. This [SO post](https://stackoverflow.com/questions/49166370/why-do-you-need-flush-at-all-if-close-is-enough) explains why they generate different output. Since this PR already modifies multiple files, I think we should update those test specs separately to use non-hardcoded results.

Reply:
Filed https://github.com/vmware-tanzu/vm-operator/issues/224 to track this.